src/interfacers/EmonHubKNXInterfacer.py

```diff
@@ -0,0 +1,273 @@
+                self.sensor[metter+"_"+dpKey] = Sensor(self.xknx, metter + "_" + dpKey, value_type=eis, group_address_state=group, always_callback=True)
+                self.xknx.devices.async_add(self.sensor[metter+"_"+dpKey])
+
+
+
+
+    def add(self, cargo):
+        self.buffer.storeItem(f)
+
+
+    async def waitSensor(self):
+        pass
+
+
+    def read(self):
+        """Read data and process
+
+        Return data as a list: [NodeID, val1, val2]
+
+        """
+        
+        interval = int(self._settings['read_interval'])
+        if time.time() - self._last_read_time < interval:
+            return
+
+        
+        self._last_read_time = time.time()
+
+        #self.displayCargo("read")
+        result = self.cargoList;
+        self.cargoList ={};
+
+        return result;
+
+    def start(self):
+        self._log.info("Start KNX interface")
+
+        task = self.loop.create_task(self.setupSensor())
+        self.loop.run_until_complete(task)
+
+        task = self.loop.create_task(self.startKnx())
+        self.loop.run_until_complete(task)
+
+
+        self.updater = self.Updater(self)
+        self.updater.start()
+
+        super().start()
+        
+
+    def set(self, **kwargs):
+        for key, setting in self._KNX_settings.items():
+            # Decide which setting value to use
+            if key in kwargs:
+                setting = kwargs[key]
+            else:
+                setting = self._KNX_settings[key]
+
+            if key in self._settings and self._settings[key] == setting:
+                continue
+            elif key == 'read_interval':
+                self._log.info("Setting %s read_interval: %s", self.name, setting)
+                self._settings[key] = float(setting)
+                continue
+            elif key == 'nodename':
+                self._log.info("Setting %s nodename: %s", self.name, setting)
+                self._settings[key] = str(setting)
+                continue
+            elif key == 'validate_checksum':
+                self._log.info("Setting %s validate_checksum: %s", self.name, setting)
+                self._settings[key] = True
+                if setting=='False':
+                    self._settings[key] = False
+                continue
+            elif key == 'meters':
+                self._log.info("Setting %s meters: %s", self.name, json.dumps(setting))
+                self._settings['meters'] = {}
+                for meter in setting:
+                    # default
+                    address = 1
+                    meter_type = "standard"
+                    records = []
+
+                    self._settings['meters'][meter] = setting[meter]
+                continue
+            else:
+                self._log.warning("'%s' is not valid for %s: %s", setting, self.name, key)
+
+        # include kwargs from parent
+        super().set(**kwargs)
+
+
+
+    class Updater(threading.Thread):
+        def __init__(self,  knxIntf):
```

Comment:
you dont give any var when you initialize the updater in self.start. What is knxIntf ?